docs/reference/analysis/charfilters/pattern-replace-charfilter.asciidoc

```diff
@@ -1,37 +1,296 @@
+      "end_offset": 2,
+      "type": "<ALPHANUM>",
+      "position": 0
+    },
+    {
+      "token": "credit",
+      "start_offset": 3,
+      "end_offset": 9,
+      "type": "<ALPHANUM>",
+      "position": 1
+    },
+    {
+      "token": "card",
+      "start_offset": 10,
+      "end_offset": 14,
+      "type": "<ALPHANUM>",
+      "position": 2
+    },
+    {
+      "token": "is",
+      "start_offset": 15,
+      "end_offset": 17,
+      "type": "<ALPHANUM>",
+      "position": 3
+    },
+    {
+      "token": "123_456_789",
+      "start_offset": 18,
+      "end_offset": 29,
+      "type": "<NUM>",
+      "position": 4
+    }
+  ]
+}
+----------------------------
+// TESTRESPONSE
+
+/////////////////////
+
+
+The above example produces the following term:
+
+[source,text]
+---------------------------
+[ My, credit, card, is 123_456_789 ]
+---------------------------
+
+
+WARNING: Using a replacement string that changes the length of the original
+text will work for search purposes, but will result in incorrect highlighting,
+as can be seen in the following example.
+
+This example inserts a space whenever it encounters a lower-case letter
+followed by an upper-case letter (i.e. `fooBarBaz` -> `foo Bar Baz`), allowing
+camelCase words to be queried individually:
+
+[source,js]
+----------------------------
+PUT my_index
+{
+  "settings": {
+    "analysis": {
+      "analyzer": {
+        "my_analyzer": {
+          "tokenizer": "standard",
+          "char_filter": [
+            "my_char_filter"
+          ],
+          "filter": [
+            "lowercase"
+          ]
+        }
+      },
+      "char_filter": {
+        "my_char_filter": {
+          "type": "pattern_replace",
+          "pattern": "(?<=\\p{Lower})(?=\\p{Upper})",
+          "replacement": " "
+        }
+      }
+    }
+  },
+  "mappings": {
+    "my_type": {
+      "properties": {
+        "text": {
+          "type": "text",
+          "analyzer": "my_analyzer"
+        }
+      }
+    }
+  }
+}
+
+POST my_index/_analyze
+{
+  "analyzer": "my_analyzer",
+  "text": "The fooBarBaz method"
+}
+----------------------------
+// CONSOLE
+
+/////////////////////
+
+[source,js]
+----------------------------
+{
+  "tokens": [
+    {
+      "token": "the",
+      "start_offset": 0,
+      "end_offset": 3,
+      "type": "<ALPHANUM>",
+      "position": 0
+    },
+    {
+      "token": "foo",
+      "start_offset": 4,
+      "end_offset": 6,
+      "type": "<ALPHANUM>",
+      "position": 1
+    },
+    {
+      "token": "bar",
+      "start_offset": 7,
+      "end_offset": 9,
+      "type": "<ALPHANUM>",
+      "position": 2
+    },
+    {
+      "token": "baz",
+      "start_offset": 10,
+      "end_offset": 13,
+      "type": "<ALPHANUM>",
+      "position": 3
+    },
+    {
+      "token": "method",
+      "start_offset": 14,
+      "end_offset": 20,
+      "type": "<ALPHANUM>",
+      "position": 4
+    }
+  ]
+}
+----------------------------
+// TEST[continued]
```

Comment:
A snippet can't be both `// TEST` and `// TESTRESPONSE`. They are (or, rather, they should be) mutually exclusive. This should just be `// TESTRESPONSE` because it isn't in sense format.